pkg/proxy/helpers.go

```diff
@@ -12,26 +15,50 @@ const (
 
 var (
 	// ClusterProxyKey is the key for the cluster proxy.
-	ClusterProxyKey = types.NamespacedName{Name: ClusterProxyName}
+	clusterProxyKey = types.NamespacedName{Name: ClusterProxyName}
 )
 
+// CheckProxyResource returns true if the deployment with the given namespacedname
```

Comment:
Nit: Did you mean to input the `namespaced` name here? If you not you might want to fix your comment.